packages/grafana-llm-app/pkg/plugin/grafana_provider.go

```diff
@@ -0,0 +1,64 @@
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type grafanaProvider struct {
+	settings LLMGatewaySettings
+	tenant   string
+	gcomKey  string
+	c        *http.Client
+}
+
+func NewGrafanaProvider(settings Settings) LLMProvider {
+	return &grafanaProvider{
+		settings: settings.LLMGateway,
+		tenant:   settings.Tenant,
+		gcomKey:  settings.GrafanaComAPIKey,
+		c: &http.Client{
+			Timeout: 2 * time.Minute,
+		},
+	}
+}
+
+func (p *grafanaProvider) Models(ctx context.Context) (ModelResponse, error) {
+	return ModelResponse{
+		Data: []ModelInfo{
+			{ID: ModelSmall},
+			{ID: ModelMedium},
+			{ID: ModelLarge},
+		},
+	}, nil
+}
```

Comment:
Does the gateway currently support these models?

Reply:
Hmm gateway should act as an OpenAI server e.g. not care about our abstractions. Our plugin backend should translate these abstractions to their respective actual model names before making the request to gateway.

Not sure what we want to do with this endpoint yet, afaik it's not in use apart from a test page 🤔 
I think we will want to fetch the actual models that are available on the API (this applies to any provider, gateway should just send it through). This way when we expose the abstraction in settings, we can provide a drop-down to the user, based on which models are available.

Reply:
Yeah makes sense, I didn't think we used it anywhere yet.